### PR TITLE
feat: validate password criteria

### PR DESCRIPTION
**Pull request by mimoham24**
# Overview

### CODE REVIEW

File: pkg/user/user.go [hunk 1]
```diff
@@ -193,3 +202,31 @@ func (u *User) SetPasswordReset(pr *PasswordReset) {
...
+
+func validatePassword(pass string) error {
+	var hasNum, hasUpper, hasLower bool
+	for _, c := range pass {
+		switch {
+		case unicode.IsNumber(c):
+			hasNum = true
+		case unicode.IsUpper(c):
+			hasUpper = true
+		case unicode.IsLetter(c) || c == ' ':
```
Comment (yk-eukarya) [inline pkg/user/user.go:214]:
```suggestion
		case unicode.IsLower(c) || c == ' ':
```

**Review by yk-eukarya — APPROVED** [review]
LGTM

File: pkg/user/user.go [hunk 2]
```diff
@@ -151,6 +156,10 @@ func (u *User) ClearAuths() {
 }
 
 func (u *User) SetPassword(pass string) error {
+	err := validatePassword(pass)
+	if err != nil {
+		return err
+	}
```
Comment (rot1024) [inline pkg/user/user.go:162]:
```suggestion
	if err := validatePassword(pass); err != nil {
		return err
	}
```

**Review by rot1024 — CHANGES_REQUESTED** [review]
Uppercase and lowercase concept of unicode are not limited to alphabet but other language characters. Is that expected?

- https://www.compart.com/en/unicode/category/Lu
- https://www.ibm.com/docs/en/i/7.2?topic=tables-unicode-lowercase-uppercase-conversion-mapping-table

**Comment by mimoham24** [comment]
@rot1024 after investigations it's ok to input a password with non-Latin characters 
the only concern is that it might cause encoding and decoding problems
so, after a meeting with @yk-eukarya we thought that having a test case with non-Latin characters will make sure that `crypto/bcrypt` can successfully encode and decode non-Latin passwords

**Comment by rot1024** [comment]
crypto/bcrypt should work. It is just a binary, so it doesn't matter at all what language the password is written in to bcrypt!

**Review by rot1024 — APPROVED** [review]

**Pull request merged**